### PR TITLE
Example notebook for torchvision dataset and batchgenerators

### PR DESCRIPTION
**Pull request by mibaumgartner**
**Checklist**
- [x] Add pull request to project (optionally delete corresponding project note)
- [x] Assign correct label (if you don't have permission to do this, someone will do it for you. 
      Please make sure to communicate the current status of the pr.)

- [x] Implementation (Starts from commit `start notebook `)
    - [x] example notebook with torchvision dataset and batchgenerators transforms
    - [x] workflow to automatically test notebooks
    - [x] introduce ToTensor and TensorOp transforms

- [x] Check if new functions are addded to `__all__` and/or `__init__` files correctly
- [x] Docstrings
- [x] Typing
- [x] Unittests (look at the line coverage of your tests, the goal is 100%!)
- [x] Update notebooks if necessary
- [x] Check test and documentation pipeline

### CODE REVIEW

File: notebooks/external_augmentation.ipynb [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
+      "\u001b[0;32m<ipython-input-1-d089e022eab3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mimport\u001b[0m \u001b[0mxy\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;31m# lets prepare a basic dataset (e.g. one from `torchvision`)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mtorchvision\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
+      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'xy'"
+     ],
+     "ename": "ModuleNotFoundError",
```
Comment (justusschock) [inline notebooks/external_augmentation.ipynb]: The state should be deleted

File: notebooks/external_augmentation.ipynb [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+     "name": "#%%\n",
+     "is_executing": false
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Integration of `Batchgenerator` transformations into the augmentation pipeline.\n",
+    "Note: when batchgenerator transformations are integrated, gradients can not be propagated through\n",
```
Comment (justusschock) [inline notebooks/external_augmentation.ipynb]: Add Note as note block ( with `>`)

Comment (justusschock) [inline notebooks/external_augmentation.ipynb]: The framework is called `batchgenerators` (we should make sure the name is spelled correct and link to it)

**Comment by mibaumgartner** [comment]
Yeah, that was my mistake... Will merge this PR after https://github.com/PhoenixDL/rising/pull/3

**Pull request merged**